File: Python3/Alternativas/ej10.py
# -*- coding: utf-8 -*-

#Algoritmo que pida los puntos centrales x1,y1,x2,y2 y los radios r1,r2 de dos 
#circunferencias y las clasifique en uno de estos estados:

#-exteriores distanciaCentros>sumaRadios
#-tangentes exteriores distanciaCentros=sumaRadios
#-secantes distanciaCentros>restaRadios
#-tangentes interiores distanciaCentros=restaRadios
#-interiores distanciaCentros<restaRadios
#-concéntricas (x1,y1)=(x2,y2)

#pedir datos
print("---[Clasificación de circunferencias]---")
print("---Circunferencia1")
x1 = float(input("coordenada X del centro> "))
y1 = float(input("coordenada Y del centro> "))
radio1 = float(input("Radio de la circunferencia> "))
print("---Circunferencia2")
x2 = float(input("coordenada X del centro> "))
y2 = float(input("coordenada Y del centro> "))
radio2 = float(input("Radio de la circunferencia> "))

#calculos
distanciaCentros = (((x2-x1)**2)+((y2-y1)**2))**(1/2)
sumaRadios = radio1 + radio2
restaRadios = radio2 - radio1

if(distanciaCentros>sumaRadios):
	print("Las circunferencias son exteriores")
if(distanciaCentros==sumaRadios):
	print("Las circunferencias son tangentes exteriores")
if(distanciaCentros>restaRadios):
	print("Las circunferencias son secantes")
if(distanciaCentros==restaRadios):
	print("Las circunferencias son tangentes interiores")
if(distanciaCentros<restaRadios):
	print("Las circunferencias son interiores")
if(distanciaCentros==0):
	print("Las circunferencias son concéntricas")
	

File: Java/Alternativas/ej10.java
import java.util.Scanner; //Importar la clase Scanner

/**
 * Algoritmo que pida los puntos centrales x1,y1,x2,y2 y los radios r1,r2 de dos 
 * circunferencias y las clasifique en uno de estos estados:
 * exteriores
 * tangentes exteriores
 * secantes
 * tangentes interiores
 * interiores
 * concéntricas
 * @author Alberto Miguel Martinez Jimenez
 */

/**
 * Algoritmo:
 * almacenar coordenadas del centro de la cirrcunferencia1
 * almacenar coordenadas del centro de la cirrcunferencia2
 * comprobar los siguentes valores:
 *
 * exteriores distanciaCentros>sumaRadios
 * tangentes exteriores distanciaCentros=sumaRadios
 * secantes distanciaCentros>restaRadios
 * tangentes interiores distanciaCentros=restaRadios
 * interiores distanciaCentros<restaRadios
 * concéntricas (x1,y1)=(x2,y2)
 *
 */


public class ej10 {

	public static void main(String[] args) {
		double x1;
		double y1;
		double x2;
		double y2;
		double r1;
		double r2;
		double distancia;
		Scanner scanner = new Scanner(System.in);//Intanciar el objeto Scanner
		
		System.out.println("Sistema para clasificar circunferencias por su posicion respecto a otra");
		System.out.println("Circunferencia1.Punto central______________________________________________________________");
		System.out.println("Introduce las coordenadas(ej.:x y)> ");
		x1 = scanner.nextDouble();
		y1 = scanner.nextDouble();
		System.out.println("Introduce el radio de la circunferencia> ");
		r1 = scanner.nextDouble();
		System.out.println("Circunferencia2.Punto central______________________________________________________________");
		System.out.println("Introduce las coordenadas(ej.:x y)>");
		x2 = scanner.nextDouble();
		y2 = scanner.nextDouble();
		System.out.println("Introduce el radio de la circunferencia> ");
		r2 = scanner.nextDouble();

		
		distancia = Math.sqrt(Math.pow((x2-x1),2)+Math.pow((y2-y1),2));//Calcular la distancia entre dos puntos cartesianos
		  
		if(distancia==(r1+r2)){
		    System.out.println("Estas circunferencias son exteriores entre si");
		}else if(distancia==(r1+r2)){
			System.out.println("Estas circunferencias son tangentes entre si");
		}else if(distancia<(r1+r2)){
			System.out.println("Estas circunferencias son exteriores entre si");
		}else if(distancia==0){
			System.out.println("Estas circunferencias son concentricas entre si");
		}else if(distancia==(r1-r2)){
			System.out.println("Estas circunferencia son tangentes interiores entre si");
		}else if(distancia<(r1-r2)){
			System.out.println("Estas circunferencias son interiores entre si");
		}
	}

}
